File: backend/src/main/java/com/project/backend/services/inter/voting/VotingService.java
package com.project.backend.services.inter.voting;

import com.project.backend.models.User;
import com.project.backend.models.enums.LevelType;
import com.project.backend.models.voting.Voting;
import org.springframework.data.domain.Page;

import java.util.List;

public interface VotingService {
    Voting create(Voting votingRequest, List<String> answer, List<Long> targetIds, long schoolId, long userId);
    Voting update(Voting votingRequest, List<String> answer, long id);
    void delete(long id);

    void deleteBy(LevelType levelType, long targetId);

    Voting findById(long id);

    List<Voting> findAllByUser(long userId);

    List<Voting> findAllByUserAndLevelClass(long userId);

    List<Voting> findAllByClass(long classId);

    Page<Voting> findAllByUser(long userId, String name, Boolean now, Boolean isNotVoted, int page, int size);
    Page<Voting> findAllByCreator(long userId, String name, Boolean now, Boolean notStarted, int page, int size);

    Page<Voting> findAllForDirector(long schoolId, long userId, String name, Boolean now, int page, int size);
    void vote(long votingId, long answerId, User user);

    void deletingUser(long userId);
}

File: frontend/src/services/base/ext/VotingService.js
import BaseService from "../BaseService";
import Cookies from "js-cookie";

class VotingService extends BaseService {
    constructor() {
        const schoolId = Cookies.get("schoolId")
        super(`/schools/${schoolId}/votings`);
    }

    createVoting(data) {
        return this.post("/create", data);
    }

    updateVoting(votingId, data) {
        return this.put(`/update/${votingId}`, data);
    }

    deleteVoting(votingId) {
        return this.delete(`/delete/${votingId}`);
    }

    getVoting(votingId) {
        return this.get(`/${votingId}`);
    }

    vote(votingId, answerId) {
        return this.post(`/${votingId}/vote/${answerId}`);
    }

    getMyVoting({page, size, name = null, now = null, isNotVote = null}) {
        const params = {
            page,
            size,
            ...(name && {name}),
            ...(now !== null && {now}),
            ...(isNotVote !== null && {isNotVote}),
        };
        return this.get("/my", {params});
    }

    getMyCreatedVoting({page, size, name = null, now = null, notStarted = null}) {
        const params = {
            page,
            size,
            ...(name && {name}),
            ...(now !== null && {now}),
            ...(notStarted !== null && {notStarted}),
        };
        return this.get("/createdByMe", {params});
    }

    getVotingForDirector({page, size, name = null, now = null}) {
        const params = {
            page,
            size,
            ...(name && {name}),
            ...(now !== null && {now}),
        };
        return this.get("/forDirector", {params});
    }
}

export default new VotingService();
